### ISSUE TITLE
Provide a way to connect a role (or role set) to a concurrent connection

### ISSUE THREAD

**Issue by syabro** [DevExpress/testcafe#7409]
### What is your Scenario?

We have a complex multi-tenant SaaS product. And we test our product using a server.
We need to make sure that every concurrent browser test is working in a separate tenant (separate Role making sure it's not used twice in the same time) to avoid errors.

### What are you suggesting?

Maybe provide a current thread/browser index through `t.testRun.browserConnection.index`?
Or update `t.testRun.browserConnection.id` that it would start with `{index}_` prefix?

Basically we need an ability to select `Role1, Role2, ..., RoleX` based on some condition provided from the test runned, making sure it's not overlaps

### Additional context

Checked this one
https://github.com/DevExpress/testcafe/issues/5356

Playwright has `workerInfo.workerIndex`
https://playwright.dev/docs/next/test-fixtures#worker-scoped-fixtures

**Comment by AlexKamaev** [comment]
I like the approach from #5356:
>I suggest simply adding concurrentInstanceId or instanceIndex to t.browser with a value unique per browser instance.

Perhaps it would be enough just to publish `t.testRun.browserConnection.id` to the `t.browser.id` property. Is it ok for you?

Please note that, currently, we have higher priority tasks. However, we can accept the PR if you help us with the implementation.

**Comment by syabro** [comment]
> Perhaps it would be enough just to publish t.testRun.browserConnection.id to the t.browser.id property. Is it ok for you?

It's a random string, isn't it? So I guess no because I can't convert it into an index

**Comment by AlexKamaev** [comment]
>It's a random string, isn't it?

Yes, the `browserConnection.id` is a random string.

I also believe that the use of the `index` property is not entirely reasonable here. Browsers run concurrently, not one after another.
You can write custom code to map the t.testRun.browserConnection.id (or t.browser.id) to indices.

**Comment by syabro** [comment]
> I also believe that the use of the index property is not entirely reasonable here. Browsers run concurrently, not one after another.
That's why I need an index from the pool, not id. I can't use one tenant in concurrent threads

> You can write custom code to map the t.testRun.browserConnection.id (or t.browser.id) to indices.
how so?

**Comment by AlexKamaev** [comment]
I still do not completely understand what the difference is between using unique browser ids and indices. Please clarify this.

**Comment by syabro** [comment]
> I still do not completely understand what the difference is between using unique browser ids and indices. Please clarify this.

Imagine you have 5 accounts: [a, b,c,d,e,f]
And 5 threads with totally random ids. How can you make sure that one and only one thread is using account at a time?

**Comment by AlexKamaev** [comment]
You can use the following code to map browser connection ids to the indices:

```JS
import { getBrowserIndex } from "./browser-indices";

fixture `f`
    .page `http://example.com`

for (let i = 0; i < 100; i++) {
    test(`test${i}`, async t => {
        console.log('browserIndex: ' + getBrowserIndex(t.testRun.browserConnection.id));
    });
}
```

```JS
const BROWSER_INDICES = {};

export function getBrowserIndex (id) {
    if (!BROWSER_INDICES[id])
        BROWSER_INDICES[id] = Object.keys(BROWSER_INDICES).length;

    return BROWSER_INDICES[id];
}
```

**Comment by syabro** [comment]
Hmmm. If I get correct and you're saying that ids will be permanent for all threads during all tests then it definitely could work. Thank you.
